main.go

```diff
@@ -0,0 +1,197 @@
+	eloteIngredients = append(eloteIngredients, "cotija cheese")
+	eloteIngredients = append(eloteIngredients, "sour cream")
+	eloteIngredients = append(eloteIngredients, "lime")
+
+	eloteRecipe := Recipe{"elote", eloteIngredients, 35}
+
+	locoMocoIngredients := make([]string, 0)
+	locoMocoIngredients = append(locoMocoIngredients, "ground beef")
+	locoMocoIngredients = append(locoMocoIngredients, "butter")
+	locoMocoIngredients = append(locoMocoIngredients, "onion")
+	locoMocoIngredients = append(locoMocoIngredients, "egg")
+	locoMocoIngredients = append(locoMocoIngredients, "bread bun")
+	locoMocoIngredients = append(locoMocoIngredients, "mushrooms")
+
+	locoMocoRecipe := Recipe{"loco moco", locoMocoIngredients, 54}
+
+	patatasBravasIngredients := make([]string, 0)
+	patatasBravasIngredients = append(patatasBravasIngredients, "potato")
+	patatasBravasIngredients = append(patatasBravasIngredients, "tomato")
+	patatasBravasIngredients = append(patatasBravasIngredients, "olive oil")
+	patatasBravasIngredients = append(patatasBravasIngredients, "onion")
+	patatasBravasIngredients = append(patatasBravasIngredients, "garlic")
+	patatasBravasIngredients = append(patatasBravasIngredients, "paprika")
+
+	patatasBravasRecipe := Recipe{"patas bravas", patatasBravasIngredients, 80}
+
+	friedRiceIngredients := make([]string, 0)
+	friedRiceIngredients = append(friedRiceIngredients, "rice")
+	friedRiceIngredients = append(friedRiceIngredients, "soy sauce")
+	friedRiceIngredients = append(friedRiceIngredients, "egg")
+	friedRiceIngredients = append(friedRiceIngredients, "onion")
+	friedRiceIngredients = append(friedRiceIngredients, "pea")
+	friedRiceIngredients = append(friedRiceIngredients, "carrot")
+	friedRiceIngredients = append(friedRiceIngredients, "sesame oil")
+
+	friedRiceRecipe := Recipe{"fried rice", friedRiceIngredients, 40}
+
+	// Create an interface of all the created recipes
+	recipes := []interface{}{eloteRecipe, locoMocoRecipe, patatasBravasRecipe, friedRiceRecipe}
+	insertManyResult, err := collection.InsertMany(context.TODO(), recipes)
+	if err != nil {
+		log.Fatal("Something went wrong trying to insert the new documents: ", err)
+	}
+	fmt.Println(insertManyResult.InsertedIDs, " documents successfully inserted.")
+
+	/*
+	 * *** FIND DOCUMENTS ***
+	 *
+	 * Now that we have data in Atlas, we can read it. To retrieve all of
+	 * the data in a collection, we create a filter for recipes that take
+	 * less than 45 minutes to prepare and sort by name (ascending)
+	 */
+
+	var filter = bson.M{"prepTimeInMinutes": bson.M{"$lt": 45}}
+	options := options.Find()
+	options.SetSort(bson.D{{"name", 1}}) // Sort by `name` field ascending
+
+	cursor, err := collection.Find(context.TODO(), filter, options)
+	if err != nil {
+		log.Fatal("Something went wrong trying to find the documents: ", err)
+	}
+
+	// Loop through the returned recipes
+	for cursor.Next(ctx) {
+		// declare a result BSON object
+		var result bson.M
+		err := cursor.Decode(&result)
```

Comment:
Instead of decoding to `bson.M`, you should be able to decode directly to your Recipes class. Here's an example:
https://stackoverflow.com/questions/51577388/in-mongodb-go-driver-how-to-marshal-unmarshal-bson-in-to-a-struct

Reply:
👍